src/hooks/useOptimisticGovernorRules.ts

```diff
@@ -0,0 +1,42 @@
+import { useState, useEffect } from "react";
+import useClient from "hooks/useOracleClient";
+import useReader from "hooks/useOracleReader";
+import { ethers } from "ethers";
+
+const ogAbi = [
+  {
+    inputs: [],
+    name: "rules",
+    outputs: [
+      {
+        internalType: "string",
+        name: "",
+        type: "string",
+      },
+    ],
+    stateMutability: "view",
+    type: "function",
+  },
+];
+
+export function useOptimisticGovernorRules() {
+  const { state } = useClient();
+  const { provider, requester } = useReader(state);
+  const [rules, setRules] = useState<undefined | string>();
+
+  useEffect(() => {
+    if (requester === undefined) return;
+    async function getRules() {
+      try {
+        const contract = new ethers.Contract(requester || "", ogAbi, provider);
+        const contractRules = await contract?.rules();
+        setRules(contractRules);
+      } catch {
+        setRules(undefined);
+      }
+    }
+    getRules();
```

Comment:
you dont really need a function, you can just move the function internals out into the use effect, since it doesnt really add anything

Reply:
I removed the function. 

For your above question on contracts that do not have rules, an error is thrown.

When removing the async function, I had to make an update to catch the promise error since it wasn’t being caught using the try catch method. Let me know if I am missing something here though. Also, the change is logging the error but let me know if you think we should remove this so it isn't being shown on all proposals without rules?

Reply:
oh got it, i missed that it was an async function, so yea you are right to use the try catch functions